pilot/pkg/xds/gen2.go

```diff
@@ -95,11 +103,15 @@ func (s *DiscoveryServer) handleCustomGenerator(con *XdsConnection, req *discove
 		return nil
 	}
 
+	// The control plane will identify itself with it's name and Istio version
+	cpIdent := fmt.Sprintf("%s~%s", podName, istioversion.Info.String())
```

Comment:
I have concerns about this because we cannot add anything in the future. I am leaning towards making this json. For example, if we want toadd the IP or something, we can easily extend that

Reply:
yeah json is better. we can add arbitrary stuff in there. Base64 encode it if you want to.

Reply:
Fixed